service/lib/dinstaller/dbus/manager.rb

```diff
@@ -41,21 +41,34 @@ def initialize(installer, logger)
         @installer = installer
         @logger = logger
 
-        # @available_base_products = installer.products
-
         super(PATH)
       end
 
       dbus_interface MANAGER_INTERFACE do
-        dbus_method :probe, "out result:u" do
-          # TODO
-          0
+        dbus_method :probe, "" do
+          # TODO: do it assynchronous. How? ractors will have problem with sharing yast data.
```

Comment:
np: asynchronous

We need to do some research here. By now, we have been using threads and, although it is far from ideal, it worked. But sure we need a better solution. Ideally, we could have an actors-like solution where every subsystem is represented by an actor and they share nothing. But that's not realistic at this point.

Reply:
sadly with way how we using yast I worry it will be very challenging